Tensile/KernelWriterAssembly.py

```diff
@@ -8919,18 +8981,34 @@ def localReadInc(self, kernel, iui, tP):
 
     if self.inTailLoop:
       inc = kernel["LocalSplitU"] * (kernel["MacroTile%s" % tP["tensorChar"]] + LdsPad) * tP["bpe"]
+      comment = " (LSU*(MT+PAD)*bpe)"
       if kernel["EnableMatrixInstruction"]:
-        if kernel["UnrollMajorLDS%s" % tP["tensorChar"]]:
-          inc = kernel["LocalSplitU"] * tP["bpe"]
-        inc *= kernel["MatrixInstK"]
+        matrixInstK = kernel["MatrixInstK"]
+        if kernel["UnrollMajorLDS%s" % tc]:
+          if kernel["DirectToLds%s" % tc] and kernel["GlobalLoadVectorWidth%c"%tc] * tP["bpe"] > 4:
+            # DirectToLds special case. Need special address coonversion
+            if (kernel["GlobalLoadVectorWidth%s"%tc] * tP["bpe"] == 8):
+              inc = 16
+            else: #if (kernel["GlobalLoadVectorWidth%s"%tc] * tP["bpe"] == 16):
+              inc = 8
+            # need to adjust incremet value if iui is odd
+            # second inc case, offset will be 64. inc should be 64 - inc
+            if iui & 1:
+              inc = 64 - inc
```

Comment:
Are the magic numbers in this section always safe to use?

Reply:
I double checked this logic with zgemm.
However, it failed.
I will debug it and provide some better logic to cover zgemm.

Reply:
Thanks for pointing this issue out.
I have submitted the fix.
I removed magic numbers.
Both DGEMM and ZGEMM works with this fix.
Please take a look.